hw/top_earlgrey/data/clocks.xdc

```diff
@@ -27,4 +27,14 @@ create_clock -add -name rv_jtag_tck -period 100.00 -waveform {0 5} [get_pin top_
 create_clock -add -name clk_spi_in  -period 100.00 -waveform {0 5} [get_pin top_*/u_spi_device/u_clk_spi_in_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
 create_clock -add -name clk_spi_out -period 100.00 -waveform {0 5} [get_pin top_*/u_spi_device/u_clk_spi_out_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
 
+## USB input delay to accommodate T_FST and the PHY's sampling logic
+## The PHY expects to only see up to one transient / fake SE0. The phase
+## relationship with the PHY's sampling clock is arbitrary, but for simplicity,
+## constrain the maximum path delay to something smaller than
+## `T_sample - T_FST(max)` to help keep the P/N skew from slipping beyond one
+## sample period.
+set_max_delay -from [get_ports {IO_UPHY_DP_RX IO_UPHY_DN_RX IO_UPHY_D_RX}] -to ${clks_48_unbuf} 6
+set_output_delay -min -clock ${clks_48_unbuf} -3 [get_ports {IO_UPHY_DP_TX IO_UPHY_DN_TX IO_UPHY_OE_N}]
+set_output_delay -max -clock ${clks_48_unbuf} 4 [get_ports {IO_UPHY_DP_TX IO_UPHY_DN_TX IO_UPHY_OE_N}] -add
```

Comment:
Oops, looks like this file is shared with chip_englishbreakfast_cw305, and it doesn't have these ports. Hm... I guess I will kick this over to pins_cw310.xdc and pins_nexysvideo.xdc.